---
layout: home
title: Eric Z Chen
---


### Adaptive non-parametric clustering  
> <sub>[[Github](https://github.com/chvlyl/kernel_kmeans_and_adaptive_clustering)]</sub>
> <sub>I implemented the adaptive non-parametric clustering method proposed by this [paper](https://arxiv.org/abs/1709.09102). For comparison, I also implemented the K-means clustering and kernel K-means clustering methods. THe three clustering methods were benchmarked on 8 simulated data as well as 8 real data. The implementaions were wrapped in to a package and a notebook of my analysis was provided. A summary report in latex was also provided. </sub>



### Udacity deep learning nano degree projects
> <sub>[[Github](https://github.com/chvlyl/udacity_deep_learning)]</sub>
> <sub>I took the Udacity deep learning nano degree program. This is an excellent program covering basics of neural networks, CNN, RNN and GAN. I finished all five deep learning projects.</sub>
> <sub> 1. Build a neural network from scratch to predict bike usge </sub>
> <sub> 2. Build a CNN model to do image classficiation based on CIFAR-10 dataset </sub>
> <sub> 3. Build a RNN model to generate Simpsons TV scripts using the Simpsons dataset of scripts from 27 seasons </sub>
> <sub> 4. Build a sequence to sequence model for translation of English and French sentence </sub>
> <sub> 5. Build a GAN model to generate new images of faces </sub>



### ZIBR: zero-inflated beta regression with random effects 
> <sub>[[R package on Github](https://github.com/chvlyl/ZIBR)][[Paper](https://doi.org/10.1093/bioinformatics/btw308)]</sub>
> <sub>The human microbial communities are associated with many human diseases such as obesity, diabetes and inflammatory bowel disease. High-throughput sequencing technology has been widely used to quantify the microbial composition in order to understand its impacts on human health. Longitudinal measurements of microbial communities are commonly obtained in many microbiome studies. A key question in such microbiome studies is to identify the microbes that are associated with clinical outcomes or environmental factors. However, microbiome compositional data are highly skewed, bounded in [0,1), and often sparse with many zeros. In addition, the observations from repeated measures in longitudinal studies are correlated. A method that takes into account these features is needed for association analysis in longitudinal microbiome data. In this paper, we propose a two-part zero-inflated Beta regression model with random effects (ZIBR) for testing the association between microbial abundance and clinical covariates for longitudinal microbiome data. The model includes a logistic regression component to model presence/absence of a microbe in the samples and a Beta regression component to model non-zero microbial abundance, where each component includes a random effect to account for the correlations among the repeated measurements on the same subject. Both simulation studies and the application to real microbiome data have shown that ZIBR model outperformed the previously used methods. The method provides a useful tool for identifying the relevant taxa based on longitudinal or repeated measures in microbiome research.</sub>

### MMDA: Multi-sample Poission model for microbiome data analysis 
> <sub> [[R package on Github](https://github.com/chvlyl/MMDA)][[Paper](https://link.springer.com/article/10.1007/s12561-016-9148-x)] </sub>
> <sub>The human microbiome, which includes the collective microbes residing in or on the human body, has a profound influence on the human health. DNA sequencing technology has made the large-scale human microbiome studies possible by using shotgun metagenomic sequencing. One important aspect of data analysis of such metagenomic data is to quantify the bacterial abundances based on the metagenomic sequencing data. Existing methods almost always quantify such abundances one sample at a time, which ignore certain systematic differences in read coverage along the genomes due to GC contents, copy number variation and the bacterial origin of replication. In order to account for such differences in read counts, we propose a multi-sample Poisson model to quantify microbial abundances based on read counts that are assigned to species-specific taxonomic markers. Our model takes into account the marker-specific effects when normalizing the sequencing count data in order to obtain more accurate quantification of the species abundances. Compared to currently available methods on simulated data and real data sets, our method has demonstrated an improved accuracy in bacterial abundance quantification, which leads to more biologically interesting results from downstream data analysis.</sub>


### For my old research projects, check [my publications](https://scholar.google.com/citations?hl=en&user=7mrZzpYAAAAJ&view_op=list_works&sortby=pubdate)


{::comment}
Add papers to Papers and export as reference list (use BMC Bioinformatics style and plain text)
{:/comment}

