<p>I start to learn QIIME for microbiome 16S rRNA data analysis. I follow this <a href="http://www.wernerlab.org/teaching/qiime/">tutorial</a> 
and the <a href="">tutorial</a> from QIIME website.</p>

<p>Step 1: install QIIME.<br />
  1. I use MacBook Pro so I download the MacQIIME <a href="http://www.wernerlab.org/software/macqiime">here</a>. Install MacQIIME.  <br />
  2. Make a copy of the tutorial data set to the home folder by:<br />
     <code>cp -R /macqiime/QIIME/qiime_tutorial ~/qiime_tutorial</code><br />
  3. Type <code>macqiime</code> to source QIIME environment variables. It is the same as a normal terminal shell, except the default 
  python is /macqiime/bin/python and there are other new QIIME-related things in system PATH. Type <code>exit</code> to go back to 
  the normal shell.</p>

<p>Step 2: install BLAST.<br />
  1. Download <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/release/2.2.22/blast-2.2.22-universal-macosx.tar.gz">BLAST</a>.<br />
  2. Unzip it by:<br />
  <code>tar -xvf blast-2.2.22-universal-macosx.tar.gz</code><br />
  3. Move it to somewhere permanent such as home directory:<br />
  <code>mv blast-2.2.22 ~/</code><br />
  4. Add the ~/blast-2.2.22/bin folder to the PATH by editing <code>~/.bash_profile</code>. Add following line (replace YourName with the actual username):<br />
  <code>export PATH=/Users/YourName/blast-2.2.22/bin:${PATH}</code><br />
  5. Type <code>blastall</code> to see if you can run it.<br />
  6. Make a <code>~/.ncbirc</code> file that points to the BLAST data directory with following content (remember to replace YourName):<br />
  <code>[NCBI]</code><br />
   <code>Data=/Users/YourName/blast-2.2.22/data/</code></p>

<p>Step 3: obtain the data.
mappling fiile.
#SampleID  BarcodeSequence LinkerPrimerSequence    Treatment DOB   Description <br />
#Example mapping file for the QIIME analysis package. These 9 samples are from a study of the effects of <br />
#exercise and diet on mouse cardiac physiology (Crawford, et al, PNAS, 2009).<br />
PC.354 AGCACGAGCCTA    YATGCTGCCTCCCGTAGGAGT   Control 20061218    Control_mouse__I.D._354<br />
PC.355 AACTCGTCGATG    YATGCTGCCTCCCGTAGGAGT   Control 20061218    Control_mouse__I.D._355<br />
PC.356 ACAGACCACTCA    YATGCTGCCTCCCGTAGGAGT   Control 20061126    Control_mouse__I.D._356</p>

<ul>
  <li>Multiple sequence alignment in QIIME:
    <ol>
      <li>You can choose from three multiple sequence alignment methods: PyNAST(Caporaso et al., 2009), MUSCLE(Edgar, 2004) and INFERNAL (Nawrocki, Kolbe, &amp; Eddy, 2009).</li>
      <li>PyNAST: need to provide a template of sequences for alignment.</li>
      <li>MUSCLE: slower than PyNAST when sequence number is large.</li>
      <li>INFERNAL: use RNA secondary structure annotation for multiple alignments.</li>
    </ol>
  </li>
  <li>Alignment filter:
    <ol>
      <li>To remove positions where gaps are in every sequence.</li>
    </ol>
  </li>
</ul>

